Automattic/woocommerce-payments
# Add state field and validation to Card Readers settings page

Fixes #3516
Fixes #3573

#### Changes proposed in this Pull Request

This PR proposes the following changes:
- Adds a new field `State` to the Card readers settings page, which is displayed if the selected country has available states.
- Adds inline validation to settings fields.

<img width="592" alt="Screenshot 2021-12-22 at 17 14 07" src="https://user-images.githubusercontent.com/8667118/147114160-522e3722-14cd-41fa-bacb-fbf11f69d28a.png">

#### Testing instructions

- Update the `_wcpay_feature_card_readers` option to 1.
- Update the `card_present_eligible` to true.
- Navigate to Payments -> Card Readers page.
- Verify the `State` field is displayed when `United States` is selected for the `Country` field.
- Verify the `phone`, `email`, `url` fields are validated.

-------------------
